### ISSUE TITLE
Undefined is not an object (evaluating 'this.map.setNativeProps') error

### ISSUE THREAD

**Issue by stesvis** [react-native-maps/react-native-maps#4242]
# Bug report

## Summary

I am using `react-native-maps` in [**Expo**](https://docs.expo.dev/versions/latest/sdk/map-view/) and I get these errors in the Logs from some **Android** users:

> TypeError
> undefined is not an object (evaluating 'this.map.setNativeProps')

![image](https://user-images.githubusercontent.com/6062684/169380848-9aee4163-10b2-4b20-85f5-e1457dc33316.png)

I can't reproduce it because it seems to always work from me in development and in production.
I know that without a repro it's basically impossible to tell, I wish I was able to reproduce it myself first but I am also "guestimating" too.

I hope that someone can at least point me to the right direction and suggest some of the **most common causes** that can lead to this error, maybe I am missing something obvious?

Maybe it has to do with how I set the `region`?

## Environment info

`react-native info` output:

```bash
$ react-native info
info Fetching system and libraries information...
(node:33608) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
    Memory: 2.17 GB / 15.68 GB
  Binaries:
    Node: 16.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.8.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.17763.0, 10.0.19041.0
  IDEs:
    Android Studio: Version     2021.1.0.0 AI-211.7628.21.2111.8139111
    Visual Studio: 16.11.32126.315 (Visual Studio Community 2019), 16.10.31515.178 (Visual Studio Enterprise 2019)
  Languages:
    Java: 1.8.0_302 - /c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.3 => 0.64.3
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps to reproduce

Unfortunately I don't know how to reproduce it. It always works for me but I see the errors in the [sentry ](https://sentry.io/)console.

### Describe what you expected to happen:

No errors I guess?

### Describe what actually happens:

> TypeError
> undefined is not an object (evaluating 'this.map.setNativeProps')

## Reproducible sample code

Custom MapView
```js
const AppMapView = ({
  initialLocation,
  style,
  visible = true,
  ...otherProps
}) => {
  if (!visible) return null;

  const location = useLastKnownLocation();
  initialLocation = initialLocation ?? location;
  const [region, setRegion] = useState();

  useEffect(() => {
    if (!initialLocation) return;

    setRegion({
      latitude: initialLocation?.latitude,
      longitude: initialLocation?.longitude,
      latitudeDelta: 3,
      longitudeDelta: 3,
    });

    return () => {};
  }, [initialLocation]);

  if (!region?.latitude || !region?.longitude) return null;

  return (
    <MapView
      initialRegion={region}
      region={region}
      //onRegionChangeComplete runs when the user stops dragging MapView
      onRegionChangeComplete={(region) => setRegion(region)}
      showsUserLocation
      showsMyLocationButton
      style={[styles.map, style]}
      // userInterfaceStyle={colorScheme}
      userInterfaceStyle="light"
      {...otherProps}>
      {otherProps.children}
    </MapView>
  );
};
```

Usage
```js
      <AppMapView initialLocation={location} {...otherProps}>
        {pins?.map((pin, index) => {
          console.log(pin);

          return (
            <Marker
              coordinate={pin}
              description={pin.description}
              key={index}
              title={pin.title}
              // onPress={() => console.log("marker pressed")}
              onCalloutPress={() => handleMarkerCalloutPress(pin)}>
              <MarkerPin height={45} width={45} />
            </Marker>
          );
        })}
      </MapView>
```

**Comment by stesvis** [comment]
PS: I've see another issue about this error, however they use it in pure RN while I see this with Expo, so the reason could be completely different...

**Comment by Simon-TechForm** [comment]
hi @stesvis 👋 Thank you for your bug report.

Seems like a strange issue. I have no explanation for the bug you are experiencing, but I'm looking to merge a TS this weekend that includes the use of useRef and a safe check for current. I would expect this to "fix" the issue you are experiencing. Just a head up 👍

**Comment by stesvis** [comment]
Hello @Simon-TechForm with Expo i am supposed to install it with the command `expo install react-native-maps` and expo is supposed to automatically grab the version that is compatible.
When I do that, it still installs v`0.30.2`. I'm not sure if I should force v`0.31.1`?

**Comment by Simon-TechForm** [comment]
hi @stesvis, sorry for the delay. I just released v1.0.0 that includes a typescript rewrite ensuring that `this.map` is present when being accessed. I can't really help you with the expo warning, but my assumption would be that it's a "use at your own risk" warning, as they didn't verify the new version. I've recently opened the discussions section on this repo, so maybe you can seek help from other expo users in there? I hope you are okay with me closing this issue for now, as this error shouldn't ever happen on the new version :)
